gemrb/core/Map.cpp

```diff
@@ -2413,11 +2416,24 @@ void Map::PlayAreaSong(int SongType, bool restart, bool hard) const
 		poi = &core->GetMusicPlaylist(pl);
 	}
 
-	if (IsStar(*poi)) return;
+	if (IsStar(*poi)) {
+		// It is currently unclear how the different games handle the change
+		// to "no music". Should GemRB stop the currently playing music?
+		// Further research is needed.
+		// At least for BG1 there is a strong assumption to just stop the
+		// music, e.g. transition from AR2616 to AR2617 (go to 2nd floor
+		// in Candlekeep Inn) or from AR2616 to AR2600 (leaving Candlekeep
+		// Inn).
+		if (isBG1) {
+			musicMgr->HardEnd();
+		}
```

Comment:
This block still makes me uneasy. Playlists have their own short interruption sounds, so they stop gracefully. This instead cuts them off hard. Does `End` really not suffice?

Reply:
I agree `HardEnd` is brute force. But `End` doesn't suffice. The very first example is Candlekeep Inn. Leaving it should stop the tavern music immediately, but with `End` it would play a whole lot longer after leaving the Inn, which is just not right.

Reply:
I see the problem. Tav2.mus refers to itself, while most do have an ending section (day1-2 have an invalid reference). So we need to fix `End` to revert to `HardEnd` in these cases. I'll look into it and you just use `End`.

Reply:
It's done.